Engine/plugin/agsplugin.cpp

```diff
@@ -1061,47 +1082,31 @@ Engine::GameInitError pl_register_plugins(const std::vector<PluginInfo> &infos)
         apl->filename = name;
         apl->savedata = info.Data;
 
-        // Compatibility with the old SnowRain module
-        if (apl->filename.CompareNoCase("ags_SnowRain20") == 0) {
-            apl->filename = "ags_snowrain";
-        }
-
-        if (apl->library.Load(apl->filename, lookup_dirs))
+        HError err = pl_load_plugin(apl, lookup_dirs);
+        if (err)
         {
-          AGS::Common::Debug::Printf(kDbgMsg_Info, "Plugin '%s' loaded from '%s', resolving imports...",
-              apl->filename.GetCStr(), apl->library.GetPath().GetCStr());
-
-          if (apl->library.GetFunctionAddress("AGS_PluginV2") == nullptr) {
-              quitprintf("Plugin '%s' is an old incompatible version.", apl->filename.GetCStr());
-          }
-          apl->engineStartup = (void(*)(IAGSEngine*))apl->library.GetFunctionAddress("AGS_EngineStartup");
-          apl->engineShutdown = (void(*)())apl->library.GetFunctionAddress("AGS_EngineShutdown");
-
-          if (apl->engineStartup == nullptr) {
-              quitprintf("Plugin '%s' is not a valid AGS plugin (no engine startup entry point)", apl->filename.GetCStr());
-          }
-          apl->onEvent = (int(*)(int,int))apl->library.GetFunctionAddress("AGS_EngineOnEvent");
-          apl->debugHook = (int(*)(const char*,int,int))apl->library.GetFunctionAddress("AGS_EngineDebugHook");
-          apl->initGfxHook = (void(*)(const char*, void*))apl->library.GetFunctionAddress("AGS_EngineInitGfx");
+            Debug::Printf(kDbgMsg_Info, "Plugin '%s' loaded from '%s', resolving imports...",
+                apl->filename.GetCStr(), apl->library.GetPath().GetCStr());
         }
         else
         {
-          String expect_filename = apl->library.GetFilenameForLib(apl->filename);
-          AGS::Common::Debug::Printf(kDbgMsg_Info, "Plugin '%s' could not be loaded (expected '%s'), trying built-in plugins...",
-              apl->filename.GetCStr(), expect_filename.GetCStr());
-          if (pl_use_builtin_plugin(apl))
-          {
-            AGS::Common::Debug::Printf(kDbgMsg_Info, "Build-in plugin '%s' found and being used.", apl->filename.GetCStr());
-          }
-          else
-          {
-            // Plugin loading has failed at this point, try using built-in plugin function stubs
-            if (RegisterPluginStubs(apl->filename.GetCStr()))
-              AGS::Common::Debug::Printf(kDbgMsg_Info, "Placeholder functions for the plugin '%s' found.", apl->filename.GetCStr());
+            String expect_filename = apl->library.GetFilenameForLib(apl->filename);
+            Debug::Printf(kDbgMsg_Error, "Plugin '%s' could not be loaded (expected '%s'):\n\t%s",
+                apl->filename.GetCStr(), expect_filename.GetCStr());
```

Comment:
Hey, this debug will crash things instantly because it's missing the last parameter.